STAC-23392: Add global configuration mode migration guide (#174)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc

kubectl create namespace suse-observability
----

[#create-values-yaml]
=== Create `values.yaml` configuration file

The `values.yaml` file contains all configuration required to deploy SUSE Observability including your license key, base URL, admin password, and sizing profile.

[NOTE]
====
*Before you continue:* Make sure you have the latest version of the Helm charts with `helm repo update`.
====

[tabs]
====
Recommended method::
+
--
Create a `values.yaml` file with the `global.suseObservability` configuration structure:

[source,yaml]
----
global:
  # Required: Set image registry for SUSE Observability images

>   imageRegistry: "registry.rancher.com"

Author: When global mode is enabled (i.e., global.suseObservability section exists), the chart automatically defaults to registry.rancher.com

  The logic:
  Scenario: Global mode enabled, not set
  global.imageRegistry: (omitted)
  Result: registry.rancher.com (auto-default)
  ────────────────────────────────────────
  Scenario: Global mode enabled, set
  global.imageRegistry: "my.registry.com"
  Result: my.registry.com
  ────────────────────────────────────────
  Scenario: Legacy mode (no global.suseObservability)
  global.imageRegistry: (omitted)
  Result: quay.io (fallback)

  imageRegistry: "registry.rancher.com"
  storageClass: "local-path" # New configuration line, the rest remains the same
----
--
====

> === Deploy SUSE Observability with Helm

Reviewer: ```suggestion
=== Deploy {stackstate-product-name} with Helm
```

[tabs]
====
Recommended method::
+
--

> To deploy SUSE Observability on Kubernetes:

Reviewer: ```suggestion
To deploy {stackstate-product-name} on Kubernetes:
```

. Before you deploy:
 ** xref:/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc#_create_namespace[Create the namespace where SUSE Observability will be installed]
 ** <<create-values-yaml,Create `values.yaml` configuration file>>

> . Deploy the latest SUSE Observability version to the `suse-observability` namespace with the following command:

Reviewer: ```suggestion
. Deploy the latest {stackstate-product-name} version to the `suse-observability` namespace with the following command:
```

 ** xref:/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc#_create_namespace[Create the namespace where SUSE Observability will be installed]
 ** <<create-values-yaml,Create `values.yaml` configuration file>>
. Deploy the latest SUSE Observability version to the `suse-observability` namespace with the following command:

[,bash]
----
helm upgrade \
  --install \
  --namespace suse-observability \
  --values local-docker-registry.yaml \
  --values values.yaml \
suse-observability \
suse-observability/suse-observability
----

--
Legacy method (Deprecated)::
+
--

[WARNING]
=====
This method is deprecated. For new installations, use the recommended method above. For existing installations using this method, see the xref:/setup/upgrade-stackstate/migrate-to-global-mode.adoc[migration guide] to transition to the new configuration format.
=====

To deploy SUSE Observability on Kubernetes:

. Before you deploy:
 ** xref:/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc#_create_namespace[Create the namespace where SUSE Observability will be installed]
 ** <<create-values-yaml,Generate `baseConfig_values.yaml` and `sizing_values.yaml`>>

> . Deploy the latest SUSE Observability version to the `suse-observability` namespace with the following command:

Reviewer: ```suggestion
To deploy {stackstate-product-name} on Kubernetes:

. Before you deploy:
 ** xref:/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc#_create_namespace[Create the namespace where {stackstate-product-name} will be installed]
 ** <<create-values-yaml,Generate `baseConfig_values.yaml` and `sizing_values.yaml`>>
. Deploy the latest {stackstate-product-name} version to the `suse-observability` namespace with the following command:
```

----

--
Legacy method (Air-gapped)::
+
--

[WARNING]
=====
This method is deprecated. For new installations, use the recommended method above. For existing installations using this method, see the xref:/setup/upgrade-stackstate/migrate-to-global-mode.adoc[migration guide] to transition to the new configuration format.
=====

> To deploy SUSE Observability in an air-gapped setup on Kubernetes:

Reviewer: ```suggestion
To deploy {stackstate-product-name} in an air-gapped setup on Kubernetes:
```

. Before you deploy:
 ** xref:/k8s-suse-rancher-prime-air-gapped.adoc[Follow these extra instructions for air-gapped installations].
 ** xref:/setup/install-stackstate/kubernetes_openshift/kubernetes_install.adoc#_create_namespace[Create the namespace where SUSE Observability will be installed]
 ** <<create-values-yaml,Generate `baseConfig_values.yaml` and `sizing_values.yaml`>>

> . Deploy the latest SUSE Observability version to the `suse-observability` namespace with the following command:

Reviewer: ```suggestion
. Deploy the latest {stackstate-product-name} version to the `suse-observability` namespace with the following command:
```
